crates/history/src/any.rs

```diff
@@ -0,0 +1,163 @@
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::browser::{BrowserHistory, BrowserLocation};
+use crate::error::HistoryResult;
+use crate::history::History;
+use crate::listener::HistoryListener;
+use crate::location::Location;
+
+/// A [`History`] that provides a universial API to the underlying history type.
+#[derive(Clone, PartialEq)]
+pub enum AnyHistory {
+    Browser(BrowserHistory),
+}
+
+/// The [`Location`] for [`AnyHistory`]
+#[derive(Clone, PartialEq)]
+pub enum AnyLocation {
+    Browser(BrowserLocation),
+}
+
+impl History for AnyHistory {
+    type Location = AnyLocation;
+
+    fn len(&self) -> usize {
+        let Self::Browser(self_) = self;
+        self_.len()
```

Comment:
Let's make the job of the next person updating this code to include another `History` a easier (and make the code, imo, more readable) by using a `match` here.